### PR TITLE
fix: dex shall show description changes

### PR DESCRIPTION
**Pull request by ishanarya0**
- firehose history should record description changes

### CODE REVIEW

File: internal/server/v1/firehose/crud.go [hunk 1]
```diff
@@ -261,7 +261,7 @@ func (api *firehoseAPI) handleUpdate(w http.ResponseWriter, r *http.Request) {
 	} else if updates.Configs.DeploymentID != existingFirehose.Configs.DeploymentID {
 		utils.WriteErr(w, errors.ErrInvalid.WithMsgf("deployment_id cannot be updated"))
 		return
-	} else if updates.Configs.KubeCluster != existingFirehose.Configs.KubeCluster {
+	} else if *(updates.Configs.KubeCluster) != *(existingFirehose.Configs.KubeCluster) {
```
Comment (mabdh) [inline internal/server/v1/firehose/crud.go:264]: let's add more test to handle this? if empty or nil

**Review by mabdh — APPROVED** [review]
